test/integration/scheduler_perf/scheduler_perf_test.go

```diff
@@ -811,13 +814,59 @@ func runWorkload(ctx context.Context, b *testing.B, tc *testCase, w *workload, c
 	numPodsScheduledPerNamespace := make(map[string]int)
 
 	if cleanup {
-		b.Cleanup(func() {
+		// This must run before controllers get cleaned up.
+		defer func() {
+			deleteNow := *metav1.NewDeleteOptions(0)
 			for namespace := range numPodsScheduledPerNamespace {
-				if err := client.CoreV1().Namespaces().Delete(ctx, namespace, metav1.DeleteOptions{}); err != nil {
-					b.Errorf("Deleting Namespace in numPodsScheduledPerNamespace: %v", err)
+				// Pods have to be deleted explicitly, with no grace period. Normally
+				// kubelet will set the DeletionGracePeriodSeconds to zero when it's okay
+				// to remove a deleted pod, but we don't run kubelet...
+				pods, err := client.CoreV1().Pods(namespace).List(ctx, metav1.ListOptions{})
+				if err != nil {
+					b.Fatalf("failed to list pods in %q: %v", namespace, err)
+				}
+				for _, pod := range pods.Items {
```

Comment:
Maybe we can reuse `CleanupPods` in `test/integration/util/util.go`, or add a new function to reuse `CleanupPods`, like `CleanupPodsInNamespace(namespace string)`.

Reply:
Agree to export the cleanup as a function which makes it more readable. The runWorkload is too long now ...

Reply:
> Not sure if we should wait until all pods to be deleted, is there any additional benefit from that?

It's required because different workloads use the same namespace. The test has wait until everything that might block deleting the namespace and the namespace itself are truly gone. If that fails, it should be a failure in the workload that failed to clean up, not a failure in the next workload which cannot create the namespace. Therefore we need to wait.

> Agree to export the cleanup as a function which makes it more readable. 

Done.